From a2cee5aaf51df9660de6df98ada6706546739d13 Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Fri, 7 Oct 2016 09:10:41 +0000
Subject: [PATCH] Compiling with GCC 6.4.0 based mingw-w64 is broken

Due to a bug in Python (https://bugs.python.org/issue11566) compiling
with a GCC 6.4.0 based mingw-w64 is broken. This patch adds a workaround
for this issue.

diff --git a/setup.py b/setup.py
index f546f1f..721b113 100644
--- a/setup.py
+++ b/setup.py
@@ -11,16 +11,16 @@ import os

 if os.name == 'nt':
     libraries = ['gmp.dll']
+    extra_compile_args=['-O3', '-fPIC', '-D_hypot=hypot']
 else:
     libraries = ['gmp']
+    extra_compile_args=['-O3', '-fPIC']

 fte_cDFA = Extension('fte.cDFA',
                      include_dirs=['fte',
                                    'thirdparty/gmp/include',
                                    ],
-                     extra_compile_args=['-O3',
-                                         '-fPIC',
-                                         ],
+                     extra_compile_args=extra_compile_args,
                      library_dirs=['thirdparty/gmp/bin',
                                    'thirdparty/gmp/lib',
                                    ],
--
2.9.3

